apps/example/lib/main.dart

```diff
@@ -1,69 +1,65 @@
-import 'package:flutter/material.dart';
+import "package:flutter/material.dart";
+import "package:flutter_version_manager/flutter_version_manager.dart";
 
 void main() {
   runApp(const MyApp());
 }
 
+/// This is the main application widget.
 class MyApp extends StatelessWidget {
+  /// The main application widget.
   const MyApp({super.key});
 
   @override
-  Widget build(BuildContext context) {
-    return MaterialApp(
-      title: 'Flutter Demo',
-      theme: ThemeData(
-        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
-        useMaterial3: true,
-      ),
-      home: const MyHomePage(title: 'Flutter Demo Home Page'),
-    );
-  }
+  Widget build(BuildContext context) => MaterialApp(
+        title: "Flutter Demo",
+        theme: ThemeData(
+          colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
+          useMaterial3: true,
+        ),
+        home: const MyHomePage(),
+      );
 }
 
+/// The home page of the application.
 class MyHomePage extends StatefulWidget {
-  const MyHomePage({super.key, required this.title});
-
-  final String title;
+  /// The home page of the application.
+  const MyHomePage({
+    super.key,
+  });
 
   @override
   State<MyHomePage> createState() => _MyHomePageState();
 }
 
 class _MyHomePageState extends State<MyHomePage> {
-  int _counter = 0;
+  VersionRepositoryService service = VersionRepositoryService();
+  String? version;
 
-  void _incrementCounter() {
-    setState(() {
-      _counter++;
+  @override
+  void initState() {
+    super.initState();
+    Future.delayed(Duration.zero, () async {
+      version = (await service.getCurrentBackendVersion()).toString();
+      setState(() {});
     });
   }
 
   @override
-  Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: AppBar(
-        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
-        title: Text(widget.title),
-      ),
-      body: Center(
-        child: Column(
-          mainAxisAlignment: MainAxisAlignment.center,
-          children: <Widget>[
-            const Text(
-              'You have pushed the button this many times:',
-            ),
-            Text(
-              '$_counter',
-              style: Theme.of(context).textTheme.headlineMedium,
+  Widget build(BuildContext context) => FlutterVersionManager(
```

Comment:
I currently don't see the update UI when i run the example app. Can you make it so it automatically prompts for an update to demonstrate the functionality working

Reply:
Done, of course there are various configurations you can try. Please fiddle with them by changing the version defined in the pubspec.yaml and the versions defined in the local implementation. Also try switching between leading (frontend / backend).